.github/workflows/windows.yml

```diff
@@ -1,71 +1,95 @@
+        python-version: [3.6]
+        build-version: [release, debug]
     steps: # each - is a new sequentially run step
       - uses: actions/checkout@master
 
       - name: Setup python
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: ${{ env.ARCH }}
 
       - name: compile
         shell: cmd
+        env:
+          CC: cl
+          CXX: cl
         run: |
-          "${{ env.VC }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          IF "${{ matrix.windows-version }}" == "windows-2016" (
+            "${{ env.VC2017 }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          )
+          IF "${{ matrix.windows-version }}" == "windows-2019" (
+            "${{ env.VC2019 }}" ${{ env.ARCH }} & python build.py ${{ matrix.build-version }}
+          )
 
       - name: upload-planner
         uses: actions/upload-artifact@master
         with:
-          name: ${{ matrix.build-version }}
-          path: builds/${{ matrix.build-version }}
+          name: compiled-planner-${{ matrix.windows-version }}-${{ matrix.build-version }}
+          path: builds/
 
 
   tests: # name of the job. Jobs run in parallel unless specified otherwise.
     name: test
-    runs-on: windows-latest
+    runs-on: ${{ matrix.windows-version }}
     needs: compile
     strategy:
       matrix:
-        python-version: ['3.6']
+        windows-version: [windows-2016, windows-2019]
+        python-version: [3.6]
     steps: # each - is a new sequentially run step
       - uses: actions/checkout@master
 
       - name: Setup python
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: ${{ env.ARCH }}
 
       - name: setup
         run: |
           pip3 install pytest tox
           mkdir builds
 
-      - name: download-planner
+      - name: download-debug
+        uses: actions/download-artifact@master
+        with:
+          name: compiled-planner-${{ matrix.windows-version }}-debug
+          path: builds/
+
+      - name: download-release
         uses: actions/download-artifact@master
         with:
-          # without 'name' attribute all artifacts are downloaded and the
-          # artifact name is used as directory name.
+          name: compiled-planner-${{ matrix.windows-version }}-release
           path: builds/
 
       - name: test
         shell: cmd
+        env:
+          CC: cl
+          CXX: cl
         run: |
           cd misc/
-          "${{ env.VC }}" ${{ env.ARCH }} & tox -e search,translator
+          IF "${{ matrix.windows-version }}" == "windows-2016" (
+            "${{ env.VC2017 }}" ${{ env.ARCH }} & tox -e search,translator
+          )
+          IF "${{ matrix.windows-version }}" == "windows-2019" (
+            "${{ env.VC2019 }}" ${{ env.ARCH }} & tox -e search,translator
```

Comment:
I'd use "tox -e py,search" in both cases.

Reply:
But how does it use the right python version then? If tox installs its own python environment?